blocks/snippets.py

```diff
@@ -0,0 +1,14 @@
+"""Routines to build frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    return tensor.sqrt(sum((t ** 2).sum() for t in tensors))
```

Comment:
It's not about literalism vs. figurativeness here, it's about not reappropriating commonplace understandings of terms to suit your own interpretation. Snippets are--quoting Wikipedia here--"literal text, or written in a simple template language", which "are a small-scale form of copy and paste programming". Hence, these helper functions are not snippets, regardless of what you name them. For many people, including myself, it will not be obvious what `snippets.py` contains. When seeing the name `expression.py` (in the context of a Theano framework) or `theano_expressions.py` I think everyone will have an immediate understanding of the kind of functions he will find inside that file.